mlflow/langchain/api_request_parallel_processor.py

```diff
@@ -171,6 +187,87 @@ def call_api(self, status_tracker: StatusTracker):
             status_tracker.increment_num_api_errors()
             status_tracker.complete_task(success=False)
 
+    def _prepare_request_for_runnable_or_chain_inference(self, request_json):
+        return APIRequest._transform_request_json_for_chat_if_necessary(request_json, self.lc_model)
+
+    @staticmethod
+    def _transform_request_json_for_chat_if_necessary(request_json, lc_model):
+        input_fields = APIRequest._get_lc_model_input_fields(lc_model)
+        if "messages" in input_fields:
+            # If the chain accepts a "messages" field directly, don't attempt to convert
+            # the request to LangChain's Message format automatically. Assume that the chain
+            # is handling the "messages" field by itself
+            return request_json
+
+        def json_dict_might_be_chat_request(json: Dict):
+            return (
+                "messages" in request_json
+                and
+                # Additional keys can't be specified when calling LangChain invoke() / batch()
+                # with chat messages
+                len(request_json) == 1
+            )
+
+        if isinstance(request_json, dict) and json_dict_might_be_chat_request(request_json):
+            try:
+                return APIRequest._convert_chat_request_or_throw(request_json)
+            except pydantic.ValidationError:
+                return request_json
+        elif isinstance(request_json, list) and all(
+            json_dict_might_be_chat_request(json) for json in request_json
+        ):
+            try:
+                return [
+                    APIRequest._convert_chat_request_or_throw(json_dict)
+                    for json_dict in request_json
+                ]
+            except pydantic.ValidationError:
+                return request_json
+        else:
+            return request_json
+
+    @staticmethod
+    def _get_lc_model_input_fields(lc_model) -> Set:
+        try:
+            if hasattr(lc_model, "input_schema") and callable(lc_model.input_schema):
+                return set(lc_model.input_schema().__fields__)
+        except Exception as e:
+            raise e
+            _logger.debug(
+                f"Unexpected exception while checking LangChain input schema for"
+                f" request transformation: {e}"
+            )
+
+        return {}
+
+    @staticmethod
+    def _convert_chat_request_or_throw(chat_request: Dict):
+        class ChatMessage(pydantic.BaseModel, extra="forbid"):
```

Comment:
Using pydantic classes for validation is easier / more robust than writing our own validation logic for chat messages / message histories. Later, we can elevate the `ChatRequest` object to a type of input to the pyfunc `predict()` method so that users can more easily write custom chat models.